### ISSUE TITLE
Maybe we can discuss on ways to improve this further maybe order the page, maybe order the list  in asc or maybe load  15 or 20 per page in a new issue?

### ISSUE THREAD

**Issue by cesswairimu** [publiclab/plots2#8542]
Maybe we can discuss on ways to improve this further maybe order the page, maybe order the list  in asc or maybe load  15 or 20 per page in a new issue?

_Originally posted by @cesswairimu in https://github.com/publiclab/plots2/issues/8115#issuecomment-707151501_

**Comment by RuthNjeri** [comment]
Yes, ordering the page in asc order is a good idea, also, I noticed that when I click the last page for Liz's followers, a notification pops up that a user has been banned, however, I didn't click on any user, just the page number(12)..this could be another issue?

![Screenshot 2020-10-12 at 17 29 06](https://user-images.githubusercontent.com/7622875/95758132-b6779600-0cb0-11eb-837c-682428e42824.png)

**Comment by cesswairimu** [comment]
I can't see that error on my end. I tried even when logged out would  you mind refreshing and see if you will still  get the error?

**Comment by RuthNjeri** [comment]
@cesswairimu I'm not seeing the error now, I think refreshing the page made it clear.

I think adding an avatar for the followers and following list is good too, also, we can add their bio if present, beneath the names..

I am curious, if a user has been banned, should they still show up as following a user or being followed?

**Comment by cesswairimu** [comment]
I agree removing banned users from the list would be ideal
I am thinking adding bio below might make the list a bit longer... maybe we could display the bio on hover?

**Comment by dldisha** [comment]
@cesswairimu I think we should add a count of the number of followers and following too in the respective tabs itself.

One more suggestion, can we add the remove follower button beside each user?

**Comment by dldisha** [comment]
One more update/suggestion that I would like to add is that instead of showing bio on hover can we show each follower/following name under their username in an italic grey text format?

**Comment by RuthNjeri** [comment]
@dldisha adding a count for followers/following is a great idea, since it is not displayed on a user's profile, unless it is shown, @cesswairimu is the followers/following link on a page somewhere or is it typed on the URL bar? I haven't seen it on this user's profile
![Screenshot 2020-10-12 at 17 54 52](https://user-images.githubusercontent.com/7622875/95760905-42d78800-0cb4-11eb-8da0-1821fbcf00df.png)

About removing a follower beside each user @dldisha, I think that button would only appear to the owner of the account.. like Liz in this case? Also, why would one want to remove a follower or is that similar to blocking a user?

**Comment by cesswairimu** [comment]
Just noting after  https://github.com/publiclab/plots2/pull/8462 the links are added on the profile page

**Comment by RuthNjeri** [comment]
> Just noting after  https://github.com/publiclab/plots2/pull/8462 the links are added on the profile page
> 
> 

I see, so the suggestion @dldisha shared on adding the count on the followers/following page is valid since it hasn't been implemented on that PR...

**Comment by stacytonui** [comment]
Suggestion: maybe reduce the column size a bit(col-6) still in the centre..so that you could add other links/activity on each side.
Yes the user avatar would be a good idea

**Comment by KarenEfereyan** [comment]
Suggestions :
 1. It might be great to allow for some padding between the user icon and the users' name.  Looks a little bit too muddled up. It might benefit from some breathing space.
2. It would be great if there could be a small span that shows the count of followers and following. Maybe just beside the following and follower text with a little left padding separating them
3. I think there should be a limit of 10users on each page to avoid top bottom scrolling. With the current implementation, it looks just perfect, all the names are showing without scrolling, but the pagination isn't. I think that should be fixed.
4. The columns look a little bit underused. Like @stacytonui mentioned, maybe the widths could be reduced and the remaining space can be used for something else.
5. On clicking the individual users' name or clicking the next page in the pagination component below, I get the message that the user is banned. I have refreshed severally but there is no change

**Comment by StlMaris123** [comment]
How about using a grid?

**Comment by 17sushmita** [comment]
These sub-tasks can be performed to enhance the Followers/Following page-

- Removing banned users from followers/following list (This generates an error notification that user has been banned.' and if we go on any other pager after it the error generated is flashed) ** high-priority
- Show count of followers/following along with Tab name `Followers` or `Following`.
- Instead of a using whole row for a user, reduce the width and use a grid or card for each user.  
- Show profile image along with follower/following name. If there is no profile image use gravatar.
- on hover to a profile, the user's bio or follower/following count shows up *not finalized.

What do you say @cesswairimu @jywarren?

I would like to try these in incremental order one by one at a time. At present, I would like to try the first one although it might take a little longer to understand the database structure how to identify if the user is banned. Any help would be appreciated.

**Comment by 17sushmita** [comment]
pinging @cesswairimu @jywarren to review the above comment. Thanks

**Comment by cesswairimu** [comment]
HI @17sushmita so sorry I missed the first ping, I love the sub tasks..you can go ahead...
a banned user has a `status: 0 `...feel free to reach out if you need more clarifications on this. Thanks

**Comment by 17sushmita** [comment]
> HI @17sushmita so sorry I missed the first ping, I love the sub tasks..you can go ahead...
> a banned user has a `status: 0 `...feel free to reach out if you need more clarifications on this. Thanks

Thank You!! will start working on it today.
